external/kafka-0-10-sql/src/test/scala/org/apache/spark/sql/kafka010/KafkaTestUtils.scala

```diff
@@ -429,7 +429,8 @@ class KafkaTestUtils(withBrokerProps: Map[String, Object] = Map.empty) extends L
   }
 
   private def waitUntilMetadataIsPropagated(topic: String, partition: Int): Unit = {
-    def isPropagated = server.apis.metadataCache.getPartitionInfo(topic, partition) match {
+    def isPropagated = server.dataPlaneRequestProcessor.metadataCache
```

Comment:
Is `dataPlaneRequestProcessor` new in Kafka 2.2? I'm wondering how much this causes a runtime incompatibility if someone were using an older Kafka client library. Not that we can't do that or require it, just wondering how essential it is. Would the current client code otherwise not work with a 2.2 broker?

Reply:
Thank you for review, @srowen . 
1. This is new, but this is `KafkaServer` API. Only our test code uses `KafkaServer`.
2. The existing client can talk to Kafka 2.2 cluster. However, in Kafka 2.2.0, there are client-side bug patches, too. Some of them will be released as Kafka 2.1.2 client library in a few month.